src/lib/Tests/EventListener/SystemInfoTabGroupListenerTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Tests\EventListener;
+
+use EzSystems\EzPlatformAdminUi\EventListener\SystemInfoTabGroupListener;
+use EzSystems\EzPlatformAdminUi\Tab\SystemInfo\SystemInfoTab;
+use EzSystems\EzPlatformAdminUi\Tab\TabGroup;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use PHPUnit\Framework\TestCase;
+use EzSystems\EzPlatformAdminUi\Tab\Event\TabEvents;
+use EzSystems\EzPlatformAdminUi\Tab\Event\TabGroupEvent;
+use EzSystems\EzPlatformAdminUi\Tab\SystemInfo\TabFactory;
+use EzSystems\EzPlatformAdminUi\Tab\TabRegistry;
+use EzSystems\EzSupportToolsBundle\SystemInfo\SystemInfoCollectorRegistry;
+
+class SystemInfoTabGroupListenerTest extends TestCase
+{
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var TabGroupEvent
+     */
+    private $event;
+
+    /**
+     * @var HttpKernelInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $httpKernel;
+
+    private $tabRegistry;
+    private $tabFactory;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->tabRegistry = $this->createMock(TabRegistry::class);
+        $this->tabFactory = $this->createMock(TabFactory::class);
+
+        $this->request = $this
+            ->getMockBuilder(Request::class)
+            ->setMethods(['getSession', 'hasSession'])
+            ->getMock();
+
+        $this->httpKernel = $this->createMock(HttpKernelInterface::class);
+        $this->event = new TabGroupEvent();
+    }
+
+    public function testOnTabGroupPreRenderWithNoSystemInfoTabGroup()
+    {
+        $systemInfoCollectorRegistry = $this->createMock(SystemInfoCollectorRegistry::class);
+        $systemInfoCollectorRegistry->expects(self::never())
+            ->method('getIdentifiers');
+
+        $systemInfoTabGroupListener = new SystemInfoTabGroupListener($this->tabRegistry, $this->tabFactory, $systemInfoCollectorRegistry);
+
+        $tabGroup = new TabGroup('some_name', []);
+        $this->event->setData($tabGroup);
+        $systemInfoTabGroupListener->onTabGroupPreRender($this->event);
+    }
+
+    /**
+     * @dataProvider dataProvider
+     */
+    public function testOnTabGroupPreRender($identifiers)
+    {
+        $tabRegistry = $this->createMock(TabRegistry::class);
+        $tabRegistry->expects(self::exactly(count($identifiers)))
+            ->method('addTab');
+        $tabFactory = $this->createMock(TabFactory::class);
+        $tabFactory->expects(self::exactly(count($identifiers)))
+            ->method('createTab')
+            ->willReturn($this->createMock(SystemInfoTab::class));
```

Comment:
As we talk on last Friday, this test will be more valuable if it will check that the proper tabs were created and registered. I'm not sure if there is the better way to write this, but following code snippet should be acceptable:


```php
foreach ($identifiers as $i => $identifier) {
    $tab = $this->createMock(SystemInfoTab::class);

    $this->tabFactory
        ->expects($this->at($i))
        ->method('createTab')
        ->with($identifier)
        ->willReturn($tab);

    $this->tabRegistry
        ->expects($this->at($i))
        ->method('addTab')
        ->with($tab, 'systeminfo');
}
```